### DISCUSSION TITLE
installing jellyseerr on rasberry Pi - Non docker environment

### DISCUSSION THREAD

**Discussion by sunnyghat** [seerr-team/seerr#302] [Q&A]
Hi, 

I wanted to see if anyone has tried to install jellyseerr on linux in a non docker container environment ?

I've followed these steps:
git clone https://github.com/Fallenbagel/jellyseerr.git && cd jellyseerr
git checkout main #if you want to run stable instead of develop
yarn install
yarn run build
yarn start

But the following commands fail to execute:
yarn install
yarn run build

**Comment by fallenbagel** [comment]
I run natively. And thats how development is done too. As for the commands failing thing, did you install yarn?

**Reply by sunnyghat** [reply]
Hi, 

I have yarn installed, error i get is:
`00h00m00s 0/0: : ERROR: [Errno 2] No such file or directory: 'install'`

this is being run in /opt/jellyseerr folder

**Reply by fallenbagel** [reply]
> Hi, 
> 
> I have yarn installed, error i get is:
> `00h00m00s 0/0: : ERROR: [Errno 2] No such file or directory: 'install'`
> 
> this is being run in /opt/jellyseerr folder

How did you install yarn? Which OS are you running

**Reply by sunnyghat** [reply]
I am trying to install on raspbian, but will be setting up arch linux in a bit to see if I get the same issue,

**Reply by fallenbagel** [reply]
It sounds like you have the cmdtest yarn installed and not actual yarn (sth that debian does thats confusing).

Run 
```
sudo apt remove cmdtest
sudo apt remove yarn
```
And then in the project directory run
```
npm install -g yarn
```

**Reply by fallenbagel** [reply]
> It sounds like you have the cmdtest yarn installed and not actual yarn (sth that debian does thats confusing).

https://stackoverflow.com/questions/65443218/yarn-run-build-no-such-file-or-directory-run

**Comment by sunnyghat** [comment]
Hi, 

Thanks, for the help, I got a little further, some additonal steps that I had to take (this was on Ubuntu) in additon to the above:
yarn install fails with the error:
`error @commitlint/cli@17.0.3: The engine "node" is incompatible with this module. Expected version ">=v14". Got "12.22.9"
error Found incompatible module.
`
to get past this I had to run the command with --ignore-engines parameter, though this does generate the following warnings:
warning " > connect-typeorm@1.1.4" has incorrect peer dependency "typeorm@^0.1.4 || ^0.2.5".
warning "react-select > @emotion/react > @emotion/babel-plugin@11.10.2" has unmet peer dependency "@babel/core@^7.0.0".
warning "react-spring > @react-spring/konva@9.5.2" has unmet peer dependency "konva@>=2.6".
warning "react-spring > @react-spring/konva@9.5.2" has unmet peer dependency "react-konva@^16.8.0  || ^17.0.0".
warning "react-spring > @react-spring/native@9.5.2" has unmet peer dependency "react-native@>=0.58".
warning "react-spring > @react-spring/three@9.5.2" has unmet peer dependency "@react-three/fiber@>=6.0".
warning "react-spring > @react-spring/three@9.5.2" has unmet peer dependency "three@>=0.126".
warning "react-spring > @react-spring/zdog@9.5.2" has unmet peer dependency "react-zdog@>=1.0".
warning "react-spring > @react-spring/zdog@9.5.2" has unmet peer dependency "zdog@>=1.0".
warning " > react-toast-notifications@2.5.1" has incorrect peer dependency "react@^16.8.0 || ^17.0.0".
warning " > react-toast-notifications@2.5.1" has incorrect peer dependency "react-dom@^16.8.0 || ^17.0.0".
warning " > @babel/cli@7.18.10" has unmet peer dependency "@babel/core@^7.0.0-0".
warning " > semantic-release-docker-buildx@1.0.1" has incorrect peer dependency "semantic-release@>=11.0.0 <18.0.0".

Once this completed and running yarn build, yarn start fails with the error:
`
root@jelly:/opt/jellyseerr# yarn start
yarn run v1.22.19
$ NODE_ENV=production node dist/index.js
/opt/jellyseerr/dist/index.js:129
            if (descriptor?.writable === true) {
                           ^

SyntaxError: Unexpected token '.'
    at wrapSafe (internal/modules/cjs/loader.js:915:16)
    at Module._compile (internal/modules/cjs/loader.js:963:27)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1027:10)
    at Module.load (internal/modules/cjs/loader.js:863:32)
    at Function.Module._load (internal/modules/cjs/loader.js:708:14)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:60:12)
    at internal/main/run_main_module.js:17:47
error Command failed with exit code 1.
`

I cannot seem to be able to get past this now.

**Reply by fallenbagel** [reply]
This has to do with your first error. You are not supposed to `--ignore-engines`. Basically first error means you have nodejs 12.22.9. You need nodejs LTS. You can use `nvm` to install nodejsLTS. It's a version problem.

**Reply by sunnyghat** [reply]
Ok managed to get nodejs LTS installed, and the service started, but the import of the libraries is failing with the error:

`2023-01-15T01:16:34.683Z [error][Jellyfin Sync]: Failed to process Jellyfin item, id: 13022{"errorMessage":"[TMDB] Failed to find media using external IMDb ID: [TMDB] Failed to find by external ID: connect ECONNREFUSED ::1:443","jellyfinitem":{"BackdropImageTags":["feb00d54da275ed36147b3ddf6011a02"],"Id":"13022","ImageTags":{"Logo":"9781828585e74694f9065db5008c22ff","Primary":"003294305f48e3473e02b7a1da7e7986"},"IsFolder":false,"MediaType":"Video","Name":"Baghban","RunTimeTicks":107039040000,"ServerId":"3afdc38d72714c6994afde65b483fa03","Type":"Movie","UserData":{"IsFavorite":false,"PlayCount":0,"PlaybackPositionTicks":0,"Played":false}}}`

I have checked and the server does have internet access

**Reply by fallenbagel** [reply]
This one is not an Internet issue but a metadata issue. Could you hop on discord so it's to discuss. There is an easy fix for this c:

**Reply by sunnyghat** [reply]
Sure will jump onto discord in a bit
